internal/benchrunner/runners/system/metrics.go

```diff
@@ -126,24 +153,98 @@ func (c *collector) collect() {
 		m.dsMetrics = dsstats
 	}
 
-	c.collectedMetrics = append(c.collectedMetrics, m)
+	return m
+}
+
+func (c *collector) publish(events [][]byte) {
+	if c.msapi == nil {
+		return
+	}
+	for _, e := range events {
+		reqBody := bytes.NewReader(e)
+		resp, err := c.msapi.Index(c.indexName(), reqBody)
+		if err != nil {
+			logger.Debugf("error indexing event: %v", err)
+			continue
+		}
```

Comment:
Do we need to close this body?
```suggestion
		}
		defer resp.Body.Close()
```

And take into account that we are keeping here a response per event. The body should be probably closed on each iteration.